cmd/minishift/cmd/ip.go

```diff
@@ -35,18 +41,33 @@ var ipCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		api := libmachine.NewClient(state.InstanceDirs.Home, state.InstanceDirs.Certs)
 		defer api.Close()
+
+		if configureAsStatic && configureAsDynamic {
+			atexit.ExitWithMessage(1, "Invalid options specified")
+		}
+
 		host, err := api.Load(constants.MachineName)
 		if err != nil {
 			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
 		}
-		ip, err := host.Driver.GetIP()
-		if err != nil {
-			atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
+
+		if configureAsDynamic {
+			minishiftNetwork.ConfigureDynamicAssignment(host.Driver)
+		} else if configureAsStatic {
+			minishiftNetwork.ConfigureStaticAssignment(host.Driver)
+		} else {
+			ip, err := minishiftNetwork.GetIP(host.Driver)
+			if err != nil {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Error getting IP: %s", err.Error()))
+			}
+			fmt.Println(ip)
 		}
-		fmt.Println(ip)
 	},
 }
 
 func init() {
+	ipCmd.Flags().BoolVar(&configureAsStatic, "static", false, "Sets the current assigned IP address as static address for the instance")
+	ipCmd.Flags().BoolVar(&configureAsDynamic, "dhcp", false, "Sets network configuration to use DHCP to assign IP address to the instance")
+
 	RootCmd.AddCommand(ipCmd)
 }
```

Comment:
@gbraad We need documentation for #1457 code changes as I am not sure what should the work flow to set the static IP.

Reply:
Static IP has not changed, as the implementation to SET only works on Hyper-V. On other hypervisors for now we can try to cset the IP address to static from the VM itself. This happens with `minishift ip --static`. This writes the config inside the VM and on restart will be re-assigned.

This however does not work for KVM, as the driver can not retrieve the information for the IP address. This logic is handled by the KVM driver... and overriding the IP is not possible without heavy modifcations to the libmachine code. I tried, and at this stage I believe this is not be best approach.

Reply:
> This however does not work for KVM, as the driver can not retrieve the information for the IP address. This logic is handled by the KVM driver... and overriding the IP is not possible without heavy modifcations to the libmachine code

Right, we must add this to docs.